core/tauri/src/lib.rs

```diff
@@ -657,25 +663,38 @@ impl<A: Assets> Context<A> {
     assets: Arc<A>,
     default_window_icon: Option<Icon>,
     app_icon: Option<Vec<u8>>,
-    system_tray_icon: Option<Icon>,
     package_info: PackageInfo,
     info_plist: (),
     pattern: Pattern,
-    #[cfg(shell_scope)] shell_scope: scope::ShellScopeConfig,
   ) -> Self {
     Self {
       config,
       assets,
       default_window_icon,
       app_icon,
-      system_tray_icon,
+      #[cfg(desktop)]
+      system_tray_icon: None,
       package_info,
       _info_plist: info_plist,
       pattern,
       #[cfg(shell_scope)]
-      shell_scope,
+      shell_scope: Default::default(),
     }
   }
+
+  /// Sets the app tray icon.
+  #[cfg(desktop)]
+  #[inline(always)]
+  pub fn with_system_tray_icon(&mut self, icon: Icon) {
+    self.system_tray_icon.replace(icon);
+  }
+
+  /// Sets the app shell scope.
+  #[cfg(shell_scope)]
+  #[inline(always)]
+  pub fn with_shell_scope(&mut self, scope: scope::ShellScopeConfig) {
```

Comment:
`with_` prefixed methods are typically used for alternative constructors, not replacements of a single field. If there is a `self` being passed to a `with_` function, then it's probably named wrong. I think `set_` in this case works best since it's completely replacing a single item.

```suggestion
  pub fn set_system_tray_icon(&mut self, icon: Icon) {
    self.system_tray_icon.replace(icon);
  }

  /// Sets the app shell scope.
  #[cfg(shell_scope)]
  #[inline(always)]
  pub fn set_shell_scope(&mut self, scope: scope::ShellScopeConfig) {
```